File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int get_height(void);

//Main method that outputs the hashes and spaces
int main(void)
{
    int height = get_height();

    //Adds a row to the pyramid
    for (int i = 1; i <= height; i++)
    {
        //Adds columns to the pyramid
        for (int j = 1; j <= height * 2 + 2; j++)
        {
            if (j > height - i && j != height + 1 && j != height + 2 && j <= height + 2 + i)
            {
                printf("#");
            }
            else if (j == height + 1 || j == height + 2 || j <= height - i)
            {
                printf(" ");
            }
        }
        printf("\n");
    }
}

//Prompting the user for the height
int get_height(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }

    while (height < 1 || height > 8);

    return height;

}

File: pset6/mario/more/mario.py
from cs50 import get_int
# Function that prompts user for a height between 1 and 8


def get_height():

    # Prompt user until heihgt is between 1 and 8
    while True:
        height = get_int("Height: ")

        if height >= 1 and height <= 8:
            return height


# Function that outputs a pyramid of the user's choice
def mario():
    # Prompt user for height
    height = get_height()

    # Loop as many times as height
    for i in range(1, height + 1):

        # Loop over all the columns in the pyramid
        for j in range(1, height * 2 + 3):

            # Print hashes
            if j > height - i and j != height + 1 and j != height + 2 and j <= height + 2 + i:
                print("#", end="")

            elif j == height + 1 or j == height + 2 or j <= height - i:
                print(" ", end="")

        # print a new line for a new row of the pyramid
        print()


mario()